examples/GstHelloWorld.hs

```diff
@@ -0,0 +1,128 @@
+{-# LANGUAGE OverloadedLabels  #-}
+{-# LANGUAGE OverloadedStrings #-}
+module Main where
+
+-- This is a translation of the Hello World gstreamer example to haskell
+-- Original C source: 
+-- https://gstreamer.freedesktop.org/documentation/application-development/basics/helloworld.html
+-- Written by Jaro Reinders on 2017-01-09
+--
+-- Most comments are copied from the source C code 
+-- with the exception of comments surrounded by [ and ]
+
+import qualified GI.GLib as GLib
+import qualified GI.Gst  as Gst
+
+import Data.GI.Base.Properties (setObjectPropertyString)
+
+import Control.Monad (void, when)
+import System.Environment
+import System.IO (stderr)
+import System.Exit (exitFailure, exitSuccess)
+import Data.Text (pack)
+import qualified Data.Text.IO as T
+import Data.Monoid ((<>))
+
+busCall :: GLib.MainLoop -> Gst.Bus -> Gst.Message -> IO Bool
+busCall loop _bus message = do
+  messageTypes <- Gst.getMessageType message
+  
+  when (Gst.MessageTypeEos `elem` messageTypes) $ do
+    putStrLn "End of stream"
+    GLib.mainLoopQuit loop
+
+  when (Gst.MessageTypeError `elem` messageTypes) $ do
+    (gerror,_debug) <- Gst.messageParseError message
+    T.hPutStrLn stderr . ("Error: " <>) =<< Gst.gerrorMessage gerror
+    GLib.mainLoopQuit loop
+  
+  return True
+
+onPadAdded :: Gst.Element -> Gst.Pad -> IO ()
+onPadAdded decoder pad = do
+  -- We can now link this pad with the vorbis-decoder sink pad
+  putStrLn "Dynamic pad created, linking demuxer/decoder"
+  Just sinkPad <- Gst.elementGetStaticPad decoder "sink"
+  void $ Gst.padLink pad sinkPad
+  Gst.objectUnref sinkPad
```

Comment:
No need to `unref`, this will be done automatically when the Haskell garbage collector collects `sinkPad`.